inst/include/interface/rcpp/rcpp_interface.hpp

```diff
@@ -0,0 +1,512 @@
+
+        //second-order derivative
+        std::shared_ptr<fims::Information<TMB_FIMS_SECOND_ORDER> > d2 =
+                fims::Information<TMB_FIMS_SECOND_ORDER>::get_instance();
+
+        std::shared_ptr<fims::SRBevertonHolt<TMB_FIMS_SECOND_ORDER> > b2 =
+                std::make_shared<fims::SRBevertonHolt<TMB_FIMS_SECOND_ORDER> >();
+
+
+        //set relative info
+        b2->id = this->id;
+        b2->steep = this->steep.value;
+        if (this->steep.estimated) {
+            if (this->steep.is_random_effect) {
+                d2->register_random_effect(b2->steep);
+            } else {
+                d2->register_parameter(b2->steep);
+            }
+        }
+        b2->rzero = this->rzero.value;
+        if (this->rzero.estimated) {
+            if (this->rzero.is_random_effect) {
+                d2->register_random_effect(b2->rzero);
+            } else {
+                d2->register_parameter(b2->rzero);
+            }
+        }
+        b2->phizero = this->phizero.value;
+        if (this->phizero.estimated) {
+            if (this->phizero.is_random_effect) {
+                d2->register_random_effect(b2->phizero);
+            } else {
+                d2->register_parameter(b2->phizero);
+            }
+        }
+        //add to Information
+        d2->recruitment_models[b2->id] = b2;
+
+
+        //third-order derivative
+        std::shared_ptr<fims::Information<TMB_FIMS_THIRD_ORDER> > d3 =
+                fims::Information<TMB_FIMS_THIRD_ORDER>::get_instance();
+
+
+        std::shared_ptr<fims::SRBevertonHolt<TMB_FIMS_THIRD_ORDER> > b3 =
+                std::make_shared<fims::SRBevertonHolt<TMB_FIMS_THIRD_ORDER> >();
+
+
+        //set relative info
+        b3->id = this->id;
+        b3->steep = this->steep.value;
+        if (this->steep.estimated) {
+            if (this->steep.is_random_effect) {
+                d3->register_random_effect(b3->steep);
+            } else {
+                d3->register_parameter(b3->steep);
+            }
+        }
+        b3->rzero = this->rzero.value;
+        if (this->rzero.estimated) {
+            if (this->rzero.is_random_effect) {
+                d3->register_random_effect(b3->rzero);
+            } else {
+                d3->register_parameter(b3->rzero);
+            }
+        }
+        b3->phizero = this->phizero.value;
+        if (this->phizero.estimated) {
+            if (this->phizero.is_random_effect) {
+                d3->register_random_effect(b3->phizero);
+            } else {
+                d3->register_parameter(b3->phizero);
+            }
+        }
+        //add to Information
+        d3->recruitment_models[b3->id] = b3;
+
+        return true;
+    }
+};
+
+/****************************************************************
+ * Selectivity Rcpp interface                                   *
+ ***************************************************************/
+class selectivity_interface_base : public fims_rcpp_interface_base {
```

Comment:
Should be SelectivityInterfaceBase

Also I have a question here - do we need one interface for each type of function (e.g. Selectivity) or does there need to be one for each module, e.g. LogisticSelectivityInterface